public_html/admin/controller/pages/sale/coupon.php

```diff
@@ -638,14 +650,29 @@ private function _getForm()
             );
         }
 
+        $this->loadModel('catalog/category');
+        $this->data['categories'] = [];
+        $allStores = $this->model_setting_store->getStores();
+        $results = $this->model_catalog_category->getCategories(0);
+        foreach ($results as $r) {
+            $this->data['categories'][$r['category_id']] = $r['name']. (count($allStores)>1 ? "   (".$r['store_name'].")":'');
+        }
+
+        $this->data['form']['fields']['category'] = $form->getFieldHtml(
+            [
+                'type'        => 'checkboxgroup',
+                'name'        => 'coupon_category[]',
+                'value'       => $this->data['coupon_category'],
+                'options'     => $this->data['categories'],
+                'style'       => 'chosen',
+                'placeholder' => $this->language->get('text_select_category'),
```

Comment:
Should we start going away from using text_   ?
It is clear that it is text

Reply:
we also have error_, entry_ and {ext_id}_ types